sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharpExplodedGraph.cs

```diff
@@ -716,18 +721,31 @@ private static IEnumerable<ProgramState> GenerateNewProgramFalseState(BinaryBran
                 : newProgramState;
 
             if (sv is ReferenceEqualsSymbolicValue referenceEqualsSymbolicValue
-                && referenceEqualsSymbolicValue.RightOperand is MemberAccessSymbolicValue rightOperand
-                && rightOperand.MemberName == "IsNullOrEmpty")
+                && referenceEqualsSymbolicValue.RightOperand is MemberAccessSymbolicValue rightOperand)
             {
-                return referenceEqualsSymbolicValue.LeftOperand.TrySetConstraint(StringConstraint.FullString, nps);       
+                if (rightOperand.MemberName == "IsNullOrEmpty")
+                {
+                    return referenceEqualsSymbolicValue.LeftOperand.TrySetConstraint(StringConstraint.FullString, nps);
+                }
+                if (rightOperand.MemberName == "IsNullOrWhiteSpace")
+                {
+                    return referenceEqualsSymbolicValue.LeftOperand.TrySetConstraint(StringConstraint.FullNotWhiteSpaceString, nps);
+                }
             }
 
             if (sv is LogicalNotSymbolicValue logicalNotSymbolicValue
                && logicalNotSymbolicValue.Operand is ReferenceEqualsSymbolicValue logicalNotReferenceEqualsSymbolicValue
-               && logicalNotReferenceEqualsSymbolicValue.RightOperand is MemberAccessSymbolicValue logicalNotRightOperand
-               && logicalNotRightOperand.MemberName == "IsNullOrEmpty")
+               && logicalNotReferenceEqualsSymbolicValue.RightOperand is MemberAccessSymbolicValue logicalNotRightOperand)
             {
-                return NullOrEmptyStringStates(nps, logicalNotReferenceEqualsSymbolicValue);
+                if (logicalNotRightOperand.MemberName == "IsNullOrEmpty")

```

Comment:
nitpick - could have a constant for `IsNullOrEmpty` and for `IsNullOrWhiteSpace`